java/pom.xml

```diff
@@ -10,6 +10,27 @@
     <packaging>pom</packaging>
 
     <name>Lance Parent</name>
+    <description>LanceDB Java API</description>
+    <url>http://lancedb.com/</url>
+
+    <developers>
+        <developer>
+            <name>Rong Rong</name>
+            <email>rong@apache.org</email>
+        </developer>
+    </developers>
+    <licenses>
+        <license>
+            <name>The Apache Software License, Version 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+        </license>
+    </licenses>
+
+    <scm>
+      <developerConnection>scm:git:git@github.com:lancedb/lance.git</developerConnection>
+      <tag>HEAD</tag>
+      <url>scm:git:git@github.com:lancedb/lance.git</url>
```

Comment:
when doing this on a branch we should make sure whoever releases it has push access to lancedb/lance

Reply:
Do we need to release from `HEAD` or we expect to release from a release branch?

also, cc @westonpace @wjones127 any concerns?

Reply:
I'm not familiar with SCM / Java releases, so not sure how this configuration works.

In Python and Rust, we create a git tag, and then release jobs are run on that specific tags commit. I think it would be nice if Java worked the same way. Talked to ChatGPT about this, and it's suggestion was to have the CI Job run a step:

```shell
tag=${GITHUB_REF#refs/tags/}
echo "Updating POM with tag ${tag}"
sed -i "s|<tag>HEAD</tag>|<tag>${tag}</tag>|g" pom.xml
```